src/js/consent-cookie.js

```diff
@@ -1,18 +1,35 @@
-const COOKIE_NAME = 'nhsuk-cookie-consent';
-const COOKIE_VERSION = 3;
-const COOKIE_LIVE_DAYS = 90;
+const COOKIE_NAME = 'cookiePreference';
+const COOKIE_DOMAIN = 'localhost'; // '.login.nhs.uk';
+const COOKIE_LIVE_DAYS = 365;
 
-function getConsentCookie() {
-  const cookie = getJSONCookie(COOKIE_NAME) || {};
-  return cookie.version < COOKIE_VERSION ? null : cookie;
+const MILLISECONDS_IN_ONE_DAY = 86400000;
+
+const DEFAULT_COOKIE_PREFERENCES = { hasSeenBanner: false, hasAcceptedOptionalCookies: true };
+const ACCEPT_ALL_COOKIE_PREFERENCES = { hasSeenBanner: true, hasAcceptedOptionalCookies: true };
+
+function getCookiePreferences() {
+  const preferenceCookie = getJSONCookie(COOKIE_NAME);
+  if (!preferenceCookie) {
+    return DEFAULT_COOKIE_PREFERENCES;
+  }
+  // previously this cookie was just set as a single flag '1'
+  if (preferenceCookie === 1) {
+    return ACCEPT_ALL_COOKIE_PREFERENCES;
+  }
+  return preferenceCookie;
 }
 
-function setConsentCookie(props) {
-  const cookieJSON = encodeURIComponent(JSON.stringify({ ...props, version: COOKIE_VERSION }));
+function setCookiePreferences(hasSeenBanner = null, hasAcceptedOptionalCookies = null) {
+  const cookiePreferences = getCookiePreferences();
+  if (hasSeenBanner) cookiePreferences.hasSeenBanner = hasSeenBanner;
```

Comment:
Do we need to make this conditional if its a boolean?

Reply:
Have pushed a change to improve the function arguments.

Reply:
I see what you mean now, you wanted named params to make it clearer and to control what is set in the cookie. That part works fine but If you pass false into `hasAcceptedOptionalCookies` then line 25 wont set it on the cookie object and it will stay as the default `true`. I see that we never call it with false so it should work fine although it is confusing.